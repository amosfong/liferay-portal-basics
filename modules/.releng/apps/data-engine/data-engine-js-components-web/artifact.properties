artifact.git.id=41a07f2549bd06fcf0eccb70babeba6d8e421b30
artifact.jsdoc.url=https://repository-cdn.liferay.com/nexus/content/repositories/liferay-public-releases/com/liferay/com.liferay.data.engine.js.components.web/1.0.143/com.liferay.data.engine.js.components.web-1.0.143-jsdoc.jar
artifact.sources.url=https://repository-cdn.liferay.com/nexus/content/repositories/liferay-public-releases/com/liferay/com.liferay.data.engine.js.components.web/1.0.143/com.liferay.data.engine.js.components.web-1.0.143-sources.jar
artifact.url=https://repository-cdn.liferay.com/nexus/content/repositories/liferay-public-releases/com/liferay/com.liferay.data.engine.js.components.web/1.0.143/com.liferay.data.engine.js.components.web-1.0.143.jar