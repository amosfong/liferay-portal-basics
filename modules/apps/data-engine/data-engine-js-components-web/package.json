{
	"dependencies": {
		"@clayui/button": "3.116.0",
		"@clayui/card": "3.123.1",
		"@clayui/data-provider": "3.119.0",
		"@clayui/drop-down": "3.123.1",
		"@clayui/empty-state": "3.116.0",
		"@clayui/form": "3.122.0",
		"@clayui/icon": "3.111.0",
		"@clayui/layout": "3.111.0",
		"@clayui/loading-indicator": "3.111.0",
		"@clayui/modal": "3.122.0",
		"@clayui/nav": "3.123.0",
		"@clayui/pagination": "3.123.1",
		"@clayui/table": "3.111.0",
		"@clayui/tabs": "3.119.0",
		"@liferay/frontend-js-react-web": "*",
		"classnames": "2.3.1",
		"dom-align": "1.12.2",
		"frontend-js-web": "*",
		"moment": "2.29.4",
		"react": "16.12.0",
		"react-dnd": "11.1.1",
		"react-dnd-html5-backend": "11.1.1",
		"react-dom": "16.12.0",
		"recharts": "^2.8.0"
	},
	"jest": {
		"collectCoverageFrom": [
			"src/main/resources/META-INF/resources/js/**/*.es.js"
		],
		"modulePathIgnorePatterns": [
			"/__mock__/",
			"/__utils__/"
		],
		"testPathIgnorePatterns": [
			"<rootDir>/test/js/[^/]+\\.js"
		]
	},
	"main": "js/index.js",
	"name": "data-engine-js-components-web",
	"private": true,
	"scripts": {
		"build": "node-scripts build",
		"checkFormat": "node-scripts format --check",
		"format": "node-scripts format",
		"test": "node-scripts test"
	},
	"version": "1.0.144"
}
