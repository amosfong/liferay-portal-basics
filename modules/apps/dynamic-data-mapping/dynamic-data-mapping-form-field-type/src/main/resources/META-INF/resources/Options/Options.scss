@import 'atlas-variables';

.add-option-button {
	border-color: $light-d2;
	border-width: 1px;
	margin-top: 16px;
	width: 100%;

	&:hover {
		border-color: $light-d2;
	}
}
