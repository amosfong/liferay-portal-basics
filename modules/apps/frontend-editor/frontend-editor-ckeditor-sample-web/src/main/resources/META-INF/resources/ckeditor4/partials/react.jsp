<%--
/**
 * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */
--%>

<%@ include file="/init.jsp" %>

<%
CKEditorSampleDisplayContext ckEditorSampleDisplayContext = (CKEditorSampleDisplayContext)request.getAttribute(CKEditorSampleWebKeys.CKEDITOR_SAMPLE_DISPLAY_CONTEXT);
%>

<react:component
	module="{CKEditor4ReactClassicEditor} from frontend-editor-ckeditor-sample-web"
	props='<%=
		HashMapBuilder.<String, Object>put(
			"editorTransformerURLs", ckEditorSampleDisplayContext.getEditorTransformerURLsJSONArray()
		).put(
			"name", "sampleReactClassicEditor"
		).put(
			"title", "Classic Editor used from a React component"
		).build()
	%>'
/>