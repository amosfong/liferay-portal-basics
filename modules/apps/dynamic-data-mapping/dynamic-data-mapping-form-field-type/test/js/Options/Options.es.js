/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

import {act, cleanup, fireEvent, render} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import {PageProvider} from 'data-engine-js-components-web';
import React from 'react';

import Options from '../../../src/main/resources/META-INF/resources/Options/Options.es';

const DEFAULT_OPTION_NAME_REGEX = /^Option[0-9]{1,}$/;

const globalLanguageDirection = Liferay.Language.direction;

const spritemap = 'icons.svg';

const OptionsWithProvider = (props) => (
	<PageProvider value={{editingLanguageId: themeDisplay.getLanguageId()}}>
		<Options {...props} />
	</PageProvider>
);

const optionsValue = {
	[themeDisplay.getLanguageId()]: [
		{
			id: 'option1',
			label: 'Option 1',
			reference: 'Option1',
			value: 'Option1',
		},
		{
			id: 'option2',
			label: 'Option 2',
			reference: 'Option2',
			value: 'Option2',
		},
	],
};

describe('Options', () => {

	// eslint-disable-next-line no-console
	const originalWarn = console.warn;

	beforeAll(() => {

		// eslint-disable-next-line no-console
		console.warn = (...args) => {
			if (/DataProvider: Trying/.test(args[0])) {
				return;
			}
			originalWarn.call(console, ...args);
		};

		Liferay.Language.direction = {
			en_US: 'rtl',
		};
	});

	afterAll(() => {

		// eslint-disable-next-line no-console
		console.warn = originalWarn;

		Liferay.Language.direction = globalLanguageDirection;
	});

	afterEach(cleanup);

	beforeEach(() => {
		jest.useFakeTimers();
		fetch.mockResponseOnce(JSON.stringify({}));
	});

	it('shows the options', () => {
		const {container, getAllByRole} = render(
			<OptionsWithProvider
				name="options"
				showKeyword={true}
				spritemap={spritemap}
				value={optionsValue}
			/>
		);

		act(() => {
			jest.runAllTimers();
		});

		const textboxes = getAllByRole('textbox');

		const referenceInputs = textboxes.filter((element) =>
			element.id.includes('keyValueReference')
		);

		expect(referenceInputs[1].value).toEqual(
			expect.stringMatching(DEFAULT_OPTION_NAME_REGEX)
		);

		referenceInputs[1].setAttribute('value', 'Any<String>');

		const optionNameInputs = textboxes.filter((element) =>
			element.id.includes('keyValueName')
		);

		expect(optionNameInputs[1].value).toEqual(
			expect.stringMatching(DEFAULT_OPTION_NAME_REGEX)
		);

		optionNameInputs[1].setAttribute('value', 'Any<String>');

		expect(container).toMatchSnapshot();
	});

	it('shows the options with not editable value', () => {
		const {getAllByRole} = render(
			<OptionsWithProvider
				keywordReadOnly={true}
				name="options"
				showKeyword={true}
				spritemap={spritemap}
				value={{
					[themeDisplay.getLanguageId()]: [
						{
							id: 'option1',
							label: 'Option 1',
							value: 'Option1',
						},
					],
				}}
			/>
		);

		act(() => {
			jest.runAllTimers();
		});

		const textboxes = getAllByRole('textbox');

		const optionNameInputs = textboxes.filter((element) =>
			element.id.includes('keyValueName')
		);

		expect(optionNameInputs[0].disabled).toBeTruthy();
		expect(optionNameInputs[0].value).toEqual('Option1');
	});

	it('shows the options with editable value', () => {
		const {getAllByRole, getByDisplayValue} = render(
			<OptionsWithProvider
				keywordReadOnly={false}
				name="options"
				onChange={jest.fn()}
				showKeyword={true}
				spritemap={spritemap}
				value={{
					[themeDisplay.getLanguageId()]: [
						{
							id: 'option1',
							label: 'Option 1',
							reference: 'Reference1',
							value: 'Option1',
						},
					],
				}}
			/>
		);

		act(() => {
			jest.runAllTimers();
		});

		userEvent.type(getByDisplayValue('Option1'), 'Option2');

		const textboxes = getAllByRole('textbox');

		const optionNameInputs = textboxes.filter((element) =>
			element.id.includes('keyValueName')
		);

		expect(optionNameInputs[0].disabled).toBeFalsy();
		expect(optionNameInputs[0].value).toEqual('Option2');
	});

	it('does show an empty option when translating', () => {
		const {container} = render(
			<OptionsWithProvider
				defaultLanguageId={themeDisplay.getLanguageId()}
				editingLanguageId="pt_BR"
				name="options"
				onChange={jest.fn()}
				spritemap={spritemap}
				value={{
					[themeDisplay.getLanguageId()]: [
						{
							id: 'option',
							label: 'Option',
							value: 'Option',
						},
					],
					pt_BR: [
						{
							id: 'option',
							label: 'Option',
							value: 'Option',
						},
					],
				}}
			/>
		);

		act(() => {
			jest.runAllTimers();
		});

		const labelInputs = container.querySelectorAll('.ddm-field-text');

		expect(labelInputs.length).toEqual(2);
	});

	it('does not changes the option value when the option label changes', () => {
		const {getAllByRole, getByDisplayValue} = render(
			<OptionsWithProvider
				name="options"
				onChange={jest.fn()}
				showKeyword={true}
				spritemap={spritemap}
				value={{
					[themeDisplay.getLanguageId()]: [
						{
							id: 'option1',
							label: 'Option 1',
							value: 'Option1',
						},
					],
				}}
			/>
		);

		act(() => {
			jest.runAllTimers();
		});

		const textboxes = getAllByRole('textbox');

		const displayNameInputs = textboxes.filter((element) =>
			element.id.includes('keyValueDisplayName')
		);

		userEvent.type(getByDisplayValue('Option 1'), 'Option 2');

		expect(displayNameInputs[0].value).toEqual('Option 2');

		const optionNameInputs = textboxes.filter((element) =>
			element.id.includes('keyValueName')
		);

		expect(optionNameInputs[0].value).toEqual('Option1');
	});

	it('edits the value of an option based on the label', () => {
		const {getAllByRole} = render(
			<OptionsWithProvider
				generateOptionValueUsingOptionLabel={true}
				name="options"
				onChange={jest.fn()}
				showKeyword={true}
				spritemap={spritemap}
				value={{
					[themeDisplay.getLanguageId()]: [
						{
							id: 'option',
							label: 'Option',
							value: 'Option',
						},
					],
				}}
			/>
		);

		act(() => {
			jest.runAllTimers();
		});

		const textboxes = getAllByRole('textbox');

		const displayNameInputs = textboxes.filter((element) =>
			element.id.includes('keyValueDisplayName')
		);

		fireEvent.change(displayNameInputs[0], {
			target: {
				value: 'Hello',
			},
		});

		act(() => {
			jest.runAllTimers();
		});

		const optionNameInputs = textboxes.filter((element) =>
			element.id.includes('keyValueName')
		);

		expect(optionNameInputs[0].value).toEqual('Hello');
	});

	it('new options are added with an unique value', () => {
		const {container, getAllByRole} = render(
			<OptionsWithProvider
				name="options"
				onChange={jest.fn()}
				showKeyword={true}
				spritemap={spritemap}
				value={{
					[themeDisplay.getLanguageId()]: [
						{
							id: 'option1',
							label: 'Option 1',
							value: 'Option1',
						},
					],
				}}
			/>
		);

		const addOptionButton = container.querySelector('.add-option-button');

		addOptionButton.click();

		act(() => {
			jest.runAllTimers();
		});

		const textboxes = getAllByRole('textbox');

		const optionNameInputs = textboxes.filter((element) =>
			element.id.includes('keyValueName')
		);

		expect(optionNameInputs[0].value).not.toEqual(
			optionNameInputs[1].value
		);
	});

	it('deduplication of value happens when the user leaves the value field', () => {
		const {getAllByRole} = render(
			<OptionsWithProvider
				name="options"
				onChange={jest.fn()}
				showKeyword={true}
				spritemap={spritemap}
				value={{
					[themeDisplay.getLanguageId()]: [
						{
							id: 'bar',
							label: 'Bar',
							value: 'Bar',
						},
						{
							id: 'foo',
							label: 'Foo',
							value: 'Foo',
						},
					],
				}}
			/>
		);

		act(() => {
			jest.runAllTimers();
		});

		const textboxes = getAllByRole('textbox');

		const optionNameInputs = textboxes.filter((element) =>
			element.id.includes('keyValueName')
		);

		fireEvent.input(optionNameInputs[1], {target: {value: 'Bar'}});

		expect(optionNameInputs[0].value).toEqual(optionNameInputs[1].value);

		fireEvent.blur(optionNameInputs[1]);

		act(() => {
			jest.runAllTimers();
		});

		expect(optionNameInputs[0].value).not.toEqual(
			optionNameInputs[1].value
		);
	});

	it.skip('adds a value to the value property when the label is empty', () => {
		const {getAllByRole} = render(
			<OptionsWithProvider
				generateOptionValueUsingOptionLabel={true}
				name="options"
				onChange={jest.fn()}
				showKeyword={true}
				spritemap={spritemap}
				value={{
					[themeDisplay.getLanguageId()]: [
						{
							id: 'bar',
							label: 'Display Name',
							reference: 'Reference',
							value: 'Name',
						},
					],
				}}
			/>
		);

		const textboxes = getAllByRole('textbox');

		const labels = textboxes.filter((element) =>
			element.id.includes('keyValueDisplayName')
		);

		fireEvent.input(labels[0], {target: {value: ''}});

		const newTextboxes = getAllByRole('textbox');

		const values = newTextboxes.filter((element) =>
			element.id.includes('keyValueName')
		);

		expect(values[0].value).toBe('Reference');
	});

	it('removes an option when click on remove button', () => {
		const {container} = render(
			<OptionsWithProvider
				defaultLanguageId={themeDisplay.getLanguageId()}
				editingLanguageId="pt_BR"
				name="options"
				onChange={jest.fn()}
				spritemap={spritemap}
				value={{
					...optionsValue,
					pt_BR: [
						{
							id: 'option1',
							label: 'Option 1',
							reference: 'Option1',
							value: 'Option1',
						},
						{
							id: 'option2',
							label: 'Option 2',
							reference: 'Option2',
							value: 'Option2',
						},
					],
				}}
			/>
		);

		let options = container.querySelectorAll('.ddm-field-options');

		expect(options.length).toEqual(2);

		const removeOptionButton = document.querySelector(
			'.ddm-option-entry .close'
		);

		fireEvent.click(removeOptionButton);

		options = container.querySelectorAll('.ddm-field-options');

		expect(options.length).toEqual(1);
	});

	it('checks if the initial value of the option reference matches the option value', () => {
		const {getAllByRole} = render(
			<OptionsWithProvider
				name="options"
				showKeyword={true}
				spritemap={spritemap}
				value={optionsValue}
			/>
		);

		const textboxes = getAllByRole('textbox');

		const referenceInputs = textboxes.filter((element) =>
			element.id.includes('keyValueReference')
		);

		expect(referenceInputs[1].value).toEqual(
			expect.stringMatching(DEFAULT_OPTION_NAME_REGEX)
		);

		const valueInputs = textboxes.filter((element) =>
			element.id.includes('keyValueName')
		);

		expect(referenceInputs[1].value).toBe(valueInputs[1].value);
	});

	describe('Normalize option reference during the onBlur event', () => {
		it('changes to the option value when the reference is duplicated', () => {
			const {getAllByRole} = render(
				<OptionsWithProvider
					name="options"
					onChange={jest.fn()}
					spritemap={spritemap}
					value={{
						[themeDisplay.getLanguageId()]: [
							{
								id: 'option1',
								label: 'Option 1',
								reference: 'Reference1',
								value: 'Option1',
							},
							{
								id: 'option2',
								label: 'Option 2',
								reference: 'Reference2',
								value: 'Option2',
							},
						],
					}}
				/>
			);

			const textboxes = getAllByRole('textbox');

			const referenceInputs = textboxes.filter((element) =>
				element.id.includes('keyValueReference')
			);

			expect(referenceInputs[0].value).toBe('Reference1');
			expect(referenceInputs[1].value).toBe('Reference2');

			fireEvent.input(referenceInputs[0], {
				target: {value: 'Reference2'},
			});

			fireEvent.blur(referenceInputs[0]);

			act(() => {
				jest.runAllTimers();
			});

			expect(referenceInputs[0].value).toBe('Option1');
			expect(referenceInputs[1].value).toBe('Reference2');
		});

		it('changes to the option value when the reference is empty', () => {
			const {getAllByRole} = render(
				<OptionsWithProvider
					name="options"
					onChange={jest.fn()}
					spritemap={spritemap}
					value={{
						[themeDisplay.getLanguageId()]: [
							{
								id: 'id',
								label: 'Label',
								reference: 'Reference',
								value: 'Value',
							},
						],
					}}
				/>
			);

			const textboxes = getAllByRole('textbox');

			const referenceInputs = textboxes.filter((element) =>
				element.id.includes('keyValueReference')
			);

			expect(referenceInputs[0].value).toBe('Reference');

			fireEvent.input(referenceInputs[0], {target: {value: ''}});

			fireEvent.blur(referenceInputs[0]);

			act(() => {
				jest.runAllTimers();
			});

			expect(referenceInputs[0].value).toEqual('Value');
		});
	});
});
