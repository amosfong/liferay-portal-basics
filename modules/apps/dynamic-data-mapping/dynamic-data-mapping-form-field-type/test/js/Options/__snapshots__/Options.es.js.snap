// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options shows the options 1`] = `
<div>
  <div
    aria-labelledby="options_fieldDetails"
    class="form-group hide"
    data-field-name="options"
  >
    <div
      class="ddm-field-options-container"
    >
      <div
        class="ddm-options-target"
      />
      <div
        class="ddm-field-options"
      >
        <span
          class="ddm-options-drag"
          draggable="true"
        >
          <svg
            class="lexicon-icon lexicon-icon-drag"
            role="presentation"
          >
            <use
              href="#drag"
            />
          </svg>
        </span>
        <div
          class="ddm-option-entry"
        >
          <div
            class="panel panel-secondary"
          >
            <button
              aria-controls="clay-id-1"
              aria-expanded="true"
              class="panel-header panel-header-link collapsable-option-header collapse-icon collapse-icon-middle show btn btn-unstyled"
              type="button"
            >
              <span
                class="panel-title"
              >
                Option 1
              </span>
              <span
                class="collapse-icon-closed"
              >
                <svg
                  class="lexicon-icon lexicon-icon-angle-right"
                  role="presentation"
                >
                  <use
                    href="#angle-right"
                  />
                </svg>
              </span>
              <span
                class="collapse-icon-open"
              >
                <svg
                  class="lexicon-icon lexicon-icon-angle-down"
                  role="presentation"
                >
                  <use
                    href="#angle-down"
                  />
                </svg>
              </span>
            </button>
            <div
              aria-labelledby="clay-id-1"
              class="panel-collapse expanded"
              id="clay-id-1"
              role="region"
            >
              <div
                class="panel-body collapsable-option-body"
              >
                <label
                  for="keyValueDisplayNameoption0"
                >
                  display-name
                </label>
                <div
                  aria-labelledby="option0_fieldDetails"
                  class="form-group hide"
                  data-field-name="option0"
                >
                  <div
                    aria-labelledby="keyValueDisplayNameoption0_fieldDetails"
                    class="form-group hide"
                    data-field-name="keyValueDisplayNameoption0"
                  >
                    <div
                      data-tooltip-align="top"
                      title="Option 1"
                    >
                      <input
                        aria-describedby="keyValueDisplayNameoption0_fieldError"
                        aria-invalid="true"
                        class="form-control ddm-field-text"
                        id="keyValueDisplayNameoption0"
                        name="keyValueDisplayNameoption0"
                        placeholder="Option"
                        type="text"
                        value="Option 1"
                      />
                    </div>
                    <input
                      name="keyValueDisplayNameoption0_edited"
                      type="hidden"
                      value="false"
                    />
                    <div
                      aria-hidden="true"
                      class="form-feedback-group lfr-de__field-feedback"
                    >
                      <div
                        aria-live="assertive"
                        class="form-feedback-item"
                        id="keyValueDisplayNameoption0_fieldError"
                      />
                    </div>
                  </div>
                  <button
                    aria-label="remove-x-option"
                    class="close close-modal"
                    type="button"
                  >
                    <svg
                      class="lexicon-icon lexicon-icon-times"
                      role="presentation"
                    >
                      <use
                        href="#times"
                      />
                    </svg>
                  </button>
                  <input
                    name="option0_edited"
                    type="hidden"
                    value="false"
                  />
                  <div
                    aria-hidden="true"
                    class="form-feedback-group lfr-de__field-feedback"
                  >
                    <div
                      aria-live="assertive"
                      class="form-feedback-item"
                      id="option0_fieldError"
                    />
                  </div>
                </div>
                <label
                  class="mt-3"
                  for="keyValueReferenceOption1"
                >
                  option-reference
                </label>
                <div
                  aria-labelledby="Option1_fieldDetails"
                  class="form-group hide"
                  data-field-name="Option1"
                >
                  <div
                    aria-labelledby="keyValueReferenceOption1_fieldDetails"
                    class="form-group hide"
                    data-field-name="keyValueReferenceOption1"
                  >
                    <div
                      data-tooltip-align="top"
                      title="Option1"
                    >
                      <input
                        aria-describedby="keyValueReferenceOption1_fieldError"
                        aria-invalid="true"
                        class="form-control ddm-field-text"
                        id="keyValueReferenceOption1"
                        name="keyValueReferenceOption1"
                        type="text"
                        value="Option1"
                      />
                    </div>
                    <input
                      name="keyValueReferenceOption1_edited"
                      type="hidden"
                      value="false"
                    />
                    <div
                      aria-hidden="true"
                      class="form-feedback-group lfr-de__field-feedback"
                    >
                      <div
                        aria-live="assertive"
                        class="form-feedback-item"
                        id="keyValueReferenceOption1_fieldError"
                      />
                    </div>
                  </div>
                  <input
                    name="Option1_edited"
                    type="hidden"
                    value="false"
                  />
                  <div
                    aria-hidden="true"
                    class="form-feedback-group lfr-de__field-feedback"
                  >
                    <div
                      aria-live="assertive"
                      class="form-feedback-item"
                      id="Option1_fieldError"
                    />
                  </div>
                </div>
                <label
                  class="mt-3"
                  for="keyValueNameOption1"
                >
                  option-name
                </label>
                <span
                  data-tooltip-align="top"
                  title="modifying-the-option-name-may-result-in-data-loss"
                >
                  Â 
                  <svg
                    class="lexicon-icon lexicon-icon-question-circle-full tooltip-icon"
                    role="presentation"
                  >
                    <use
                      href="#question-circle-full"
                    />
                  </svg>
                </span>
                <div
                  aria-labelledby="Option1_fieldDetails"
                  class="form-group hide"
                  data-field-name="Option1"
                >
                  <div
                    aria-labelledby="keyValueNameOption1_fieldDetails"
                    class="form-group hide"
                    data-field-name="keyValueNameOption1"
                  >
                    <div
                      data-tooltip-align="top"
                      title="Option1"
                    >
                      <input
                        aria-describedby="keyValueNameOption1_fieldError"
                        aria-invalid="true"
                        class="form-control ddm-field-text"
                        id="keyValueNameOption1"
                        name="keyValueNameOption1"
                        type="text"
                        value="Option1"
                      />
                    </div>
                    <input
                      name="keyValueNameOption1_edited"
                      type="hidden"
                      value="false"
                    />
                    <div
                      aria-hidden="true"
                      class="form-feedback-group lfr-de__field-feedback"
                    >
                      <div
                        aria-live="assertive"
                        class="form-feedback-item"
                        id="keyValueNameOption1_fieldError"
                      />
                    </div>
                  </div>
                  <input
                    name="Option1_edited"
                    type="hidden"
                    value="false"
                  />
                  <div
                    aria-hidden="true"
                    class="form-feedback-group lfr-de__field-feedback"
                  >
                    <div
                      aria-live="assertive"
                      class="form-feedback-item"
                      id="Option1_fieldError"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ddm-options-target"
      />
      <div
        class="ddm-field-options"
      >
        <span
          class="ddm-options-drag"
          draggable="true"
        >
          <svg
            class="lexicon-icon lexicon-icon-drag"
            role="presentation"
          >
            <use
              href="#drag"
            />
          </svg>
        </span>
        <div
          class="ddm-option-entry"
        >
          <div
            class="panel panel-secondary"
          >
            <button
              aria-controls="clay-id-2"
              aria-expanded="true"
              class="panel-header panel-header-link collapsable-option-header collapse-icon collapse-icon-middle show btn btn-unstyled"
              type="button"
            >
              <span
                class="panel-title"
              >
                Option 2
              </span>
              <span
                class="collapse-icon-closed"
              >
                <svg
                  class="lexicon-icon lexicon-icon-angle-right"
                  role="presentation"
                >
                  <use
                    href="#angle-right"
                  />
                </svg>
              </span>
              <span
                class="collapse-icon-open"
              >
                <svg
                  class="lexicon-icon lexicon-icon-angle-down"
                  role="presentation"
                >
                  <use
                    href="#angle-down"
                  />
                </svg>
              </span>
            </button>
            <div
              aria-labelledby="clay-id-2"
              class="panel-collapse expanded"
              id="clay-id-2"
              role="region"
            >
              <div
                class="panel-body collapsable-option-body"
              >
                <label
                  for="keyValueDisplayNameoption1"
                >
                  display-name
                </label>
                <div
                  aria-labelledby="option1_fieldDetails"
                  class="form-group hide"
                  data-field-name="option1"
                >
                  <div
                    aria-labelledby="keyValueDisplayNameoption1_fieldDetails"
                    class="form-group hide"
                    data-field-name="keyValueDisplayNameoption1"
                  >
                    <div
                      data-tooltip-align="top"
                      title="Option 2"
                    >
                      <input
                        aria-describedby="keyValueDisplayNameoption1_fieldError"
                        aria-invalid="true"
                        class="form-control ddm-field-text"
                        id="keyValueDisplayNameoption1"
                        name="keyValueDisplayNameoption1"
                        placeholder="Option"
                        type="text"
                        value="Option 2"
                      />
                    </div>
                    <input
                      name="keyValueDisplayNameoption1_edited"
                      type="hidden"
                      value="false"
                    />
                    <div
                      aria-hidden="true"
                      class="form-feedback-group lfr-de__field-feedback"
                    >
                      <div
                        aria-live="assertive"
                        class="form-feedback-item"
                        id="keyValueDisplayNameoption1_fieldError"
                      />
                    </div>
                  </div>
                  <button
                    aria-label="remove-x-option"
                    class="close close-modal"
                    type="button"
                  >
                    <svg
                      class="lexicon-icon lexicon-icon-times"
                      role="presentation"
                    >
                      <use
                        href="#times"
                      />
                    </svg>
                  </button>
                  <input
                    name="option1_edited"
                    type="hidden"
                    value="false"
                  />
                  <div
                    aria-hidden="true"
                    class="form-feedback-group lfr-de__field-feedback"
                  >
                    <div
                      aria-live="assertive"
                      class="form-feedback-item"
                      id="option1_fieldError"
                    />
                  </div>
                </div>
                <label
                  class="mt-3"
                  for="keyValueReferenceOption2"
                >
                  option-reference
                </label>
                <div
                  aria-labelledby="Option2_fieldDetails"
                  class="form-group hide"
                  data-field-name="Option2"
                >
                  <div
                    aria-labelledby="keyValueReferenceOption2_fieldDetails"
                    class="form-group hide"
                    data-field-name="keyValueReferenceOption2"
                  >
                    <div
                      data-tooltip-align="top"
                      title="Option2"
                    >
                      <input
                        aria-describedby="keyValueReferenceOption2_fieldError"
                        aria-invalid="true"
                        class="form-control ddm-field-text"
                        id="keyValueReferenceOption2"
                        name="keyValueReferenceOption2"
                        type="text"
                        value="Any<String>"
                      />
                    </div>
                    <input
                      name="keyValueReferenceOption2_edited"
                      type="hidden"
                      value="false"
                    />
                    <div
                      aria-hidden="true"
                      class="form-feedback-group lfr-de__field-feedback"
                    >
                      <div
                        aria-live="assertive"
                        class="form-feedback-item"
                        id="keyValueReferenceOption2_fieldError"
                      />
                    </div>
                  </div>
                  <input
                    name="Option2_edited"
                    type="hidden"
                    value="false"
                  />
                  <div
                    aria-hidden="true"
                    class="form-feedback-group lfr-de__field-feedback"
                  >
                    <div
                      aria-live="assertive"
                      class="form-feedback-item"
                      id="Option2_fieldError"
                    />
                  </div>
                </div>
                <label
                  class="mt-3"
                  for="keyValueNameOption2"
                >
                  option-name
                </label>
                <span
                  data-tooltip-align="top"
                  title="modifying-the-option-name-may-result-in-data-loss"
                >
                  Â 
                  <svg
                    class="lexicon-icon lexicon-icon-question-circle-full tooltip-icon"
                    role="presentation"
                  >
                    <use
                      href="#question-circle-full"
                    />
                  </svg>
                </span>
                <div
                  aria-labelledby="Option2_fieldDetails"
                  class="form-group hide"
                  data-field-name="Option2"
                >
                  <div
                    aria-labelledby="keyValueNameOption2_fieldDetails"
                    class="form-group hide"
                    data-field-name="keyValueNameOption2"
                  >
                    <div
                      data-tooltip-align="top"
                      title="Option2"
                    >
                      <input
                        aria-describedby="keyValueNameOption2_fieldError"
                        aria-invalid="true"
                        class="form-control ddm-field-text"
                        id="keyValueNameOption2"
                        name="keyValueNameOption2"
                        type="text"
                        value="Any<String>"
                      />
                    </div>
                    <input
                      name="keyValueNameOption2_edited"
                      type="hidden"
                      value="false"
                    />
                    <div
                      aria-hidden="true"
                      class="form-feedback-group lfr-de__field-feedback"
                    >
                      <div
                        aria-live="assertive"
                        class="form-feedback-item"
                        id="keyValueNameOption2_fieldError"
                      />
                    </div>
                  </div>
                  <input
                    name="Option2_edited"
                    type="hidden"
                    value="false"
                  />
                  <div
                    aria-hidden="true"
                    class="form-feedback-group lfr-de__field-feedback"
                  >
                    <div
                      aria-live="assertive"
                      class="form-feedback-item"
                      id="Option2_fieldError"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        class="add-option-button btn btn-secondary"
        type="button"
      >
        <span>
          add-option
        </span>
      </button>
    </div>
    <input
      name="options_edited"
      type="hidden"
      value="false"
    />
    <div
      aria-hidden="true"
      class="form-feedback-group lfr-de__field-feedback"
    >
      <div
        aria-live="assertive"
        class="form-feedback-item"
        id="options_fieldError"
      />
    </div>
  </div>
</div>
`;
